rust-runtime/aws-smithy-http-server/examples/pokemon-service/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Pokémon Service
+//!
+//! This crate implements the Pokémon Service.
+#![warn(missing_docs, missing_debug_implementations, rust_2018_idioms)]
+use std::{
+    collections::HashMap,
+    convert::TryInto,
+    sync::{atomic::AtomicU64, Arc},
+};
+
+use aws_smithy_http_server::Extension;
+use pokemon_service_sdk::{error, input, model, output};
+use tracing::{debug, error};
+use tracing_subscriber::{prelude::*, EnvFilter};
+
+const PIKACHU_ENGLISH_FLAVOR_TEXT: &str =
+    "When several of these Pokémon gather, their electricity could build and cause lightning storms.";
+const PIKACHU_SPANISH_FLAVOR_TEXT: &str =
+    "Cuando varios de estos Pokémon se juntan, su energía puede causar fuertes tormentas.";
+const PIKACHU_ITALIAN_FLAVOR_TEXT: &str =
+    "Quando vari Pokémon di questo tipo si radunano, la loro energia può causare forti tempeste.";
+
+/// Setup `tracing::subscriber` to read the log level from RUST_LOG environment variable.
+pub fn setup_tracing() {
+    let format = tracing_subscriber::fmt::layer()
+        .with_ansi(true)
+        .with_line_number(true)
+        .with_level(true);
+    let filter = EnvFilter::try_from_default_env()
+        .or_else(|_| EnvFilter::try_new("info"))
+        .unwrap();
+    tracing_subscriber::registry().with(format).with(filter).init();
+}
+
+/// Structure holding the translations for a Pokémon description.
+#[derive(Debug)]
+struct PokemonTranslations {
+    en: String,
+    es: String,
+    it: String,
+}
+
+impl PokemonTranslations {
+    /// Create a new translations entry for a Pokémon.
+    pub fn new(en: &str, es: &str, it: &str) -> Self {
```

Comment:
```suggestion
    pub fn new(en: String, es: String, it: String) -> Self {
```

https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control

Reply:
Alternatively, just remove the constructor, it's not really adding much value to such a simple example.

Reply:
I think it's probably better to just remove `new()`